task/k8s/resources/resource_job.go

```diff
@@ -0,0 +1,347 @@
+				Spec: kubernetes_core.PodSpec{
+					TerminationGracePeriodSeconds: &jobTerminationGracePeriod,
+					ActiveDeadlineSeconds:         &jobActiveDeadlineSeconds,
+					// We don't want pods to restart if the container exits with a non–zero status.
+					// Only when there is a pod failure.
+					RestartPolicy: kubernetes_core.RestartPolicyNever,
+					NodeSelector:  jobNodeSelector,
+					Containers: []kubernetes_core.Container{
+						{
+							Name:  j.Identifier,
+							Image: j.Attributes.Task.Environment.Image,
+							Resources: kubernetes_core.ResourceRequirements{
+								Limits: jobLimits,
+								Requests: kubernetes_core.ResourceList{
+									// Don't allocate any resources statically and let the pod scale vertically when and if required.
+									kubernetes_core.ResourceMemory: kubernetes_resource.MustParse("0"),
+									kubernetes_core.ResourceCPU:    kubernetes_resource.MustParse("0"),
+								},
+							},
+							Command: []string{
+								"sh", "-c", script,
+							},
+							Env:          jobEnvironment,
+							VolumeMounts: jobVolumeMounts,
+						},
+					},
+					Volumes: jobVolumes,
+				},
+			},
+		},
+	}
+
+	// Ask Kubernetes to create the job.
+	log.Printf("[INFO] Creating new Job: %#v", job)
+	out, err := j.Client.Services.Batch.Jobs(j.Client.Namespace).Create(ctx, &job, kubernetes_meta.CreateOptions{})
+	if err != nil {
+		if statusErr, ok := err.(*kubernetes_errors.StatusError); ok && statusErr.ErrStatus.Code == 409 {
+			return j.Read(ctx)
+		}
+		return err
+	}
+	log.Printf("[INFO] Submitted new job: %#v", out)
+
+	j.Resource = out
+	return nil
+}
+
+func (j *Job) Read(ctx context.Context) error {
+	job, err := j.Client.Services.Batch.Jobs(j.Client.Namespace).Get(ctx, j.Identifier, kubernetes_meta.GetOptions{})
+	if err != nil {
+		if statusErr, ok := err.(*kubernetes_errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
+			return universal.NotFoundError
+		}
+		return err
+	}
+	eventListOptions := kubernetes_meta.ListOptions{FieldSelector: fields.OneTermEqualSelector("involvedObject.name", job.Name).String()}
+	events, err := j.Client.Services.Core.Events(j.Client.Namespace).List(ctx, eventListOptions)
+	if err != nil {
+		return err
+	}
+	for _, event := range events.Items {
+		j.Attributes.Events = append(j.Attributes.Events, universal.Event{
+			Time: event.FirstTimestamp.Time,
+			Code: event.Message,
+			Description: []string{
+				event.Reason,
+				event.Action,
+			},
+		})
+	}
+	j.Attributes.Status = map[string]int{
+		"active":    int(job.Status.Active),
+		"succeeded": int(job.Status.Succeeded),
+		"failed":    int(job.Status.Failed),
+	}
+	j.Resource = job
+	return nil
+}
+
+func (j *Job) Delete(ctx context.Context) error {
+	_, err := j.Client.Services.Batch.Jobs(j.Client.Namespace).Get(ctx, j.Identifier, kubernetes_meta.GetOptions{})
+	if err != nil {
+		if statusErr, ok := err.(*kubernetes_errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
+			return nil
```

Comment:
Avoidable is check is !=404

I also would create smaller checks to avoid half of the Quijote in one sentence
```
if statusErr, ok := err.(*kubernetes_errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
```

Reply:
Not avoidable in this case: if the resource does not exist, the execution can't continue either.

Reply:
?

Reply:
Continued in https://github.com/iterative/terraform-provider-iterative/pull/237/files/739955bf1943bb25a1d3711ce232f4d5ff67676b#r750394337